tests/lambdas/leverDifferent.test.js

```diff
@@ -0,0 +1,136 @@
+const lambda = require('../../src/functions/lever')
+const got = require('got')
+const url = require('url')
+
+const leverBody = {
+  allLever: {
+    group: [
+      {
+        edges: [
+          {
+            node: {
+              lever_id: 'cd839b3d-d68d-4d21-a68b-8038479d5c6e'
+            }
+          }
+        ]
+      },
+      {
+        edges: [
+          {
+            node: {
+              lever_id: '3f8dc4d0-a310-4e8e-a637-51e9b5d7fd80'
+            }
+          },
+          {
+            node: {
+              lever_id: 'e4180acb-8125-404c-adaa-8bc7b7a7b354'
+            }
+          },
+          {
+            node: {
+              lever_id: 'f8066e39-8d0e-4759-b27b-8898140ec565'
+            }
+          },
+          {
+            node: {
+              lever_id: '2926b1d1-7ebb-43b7-86dd-793ec44c34ae'
+            }
+          },
+          {
+            node: {
+              lever_id: 'faef66f9-1ac1-4f1e-ac7e-102e9dd8d6ae'
+            }
+          }
+        ]
+      },
+      {
+        edges: [
+          {
+            node: {
+              lever_id: '2dc54cf4-bbed-46c6-86bc-3042f6016432'
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+
+jest.mock('got')
+jest.mock('url')
+
+describe('different data', () => {
+  let response
+
+  beforeAll(async () => {
+    url.URL.mockReturnValueOnce({
+      href: 'http://localhost:8000/meta.json',
+      origin: 'http://localhost:8000',
+      protocol: 'http:',
+      username: '',
+      password: '',
+      host: 'localhost:8000',
+      hostname: 'localhost',
+      port: '8000',
+      pathname: '/meta.json',
+      search: '',
+      hash: ''
+    })
+
+    url.URL.mockReturnValueOnce({
+      href: 'https://api.lever.co/v0/postings/yld?mode=json',
+      origin: 'https://api.lever.co',
+      protocol: 'https:',
+      username: '',
+      password: '',
+      host: 'api.lever.co',
+      hostname: 'api.lever.co',
+      port: '',
+      pathname: '/v0/postings/yld',
+      search: '?mode=json',
+      hash: ''
+    })
+
+    got.mockReturnValueOnce({
+      body: {
+        allJobIds: ['one', 'two', 'three']
+      }
+    })
+
+    got.mockReturnValueOnce({
+      body: [
+        {
+          id: 'one'
+        },
+        {
+          id: 'two'
+        },
+        {
+          id: 'three'
+        },
+        {
+          id: 'four'
+        },
+        {
+          id: 'five'
+        }
+      ]
+    })
+
+    got.post.mockReturnValueOnce({ body: leverBody })
+
+    response = await lambda.handler({
+      headers: {
+        authorization: 'basic foo_test_thingy'
+      }
+    })
+  })
+
+  it('Calls the lambda leaver webhook', () => {
+    expect(got.post).toHaveBeenCalledTimes(1)
+  })
+
+  it('returns the response body from the lever webhook', () => {
+    expect(response.body).toEqual(leverBody)
```

Comment:
I think it's fine. I'm still trying to understand how the whole Lever lambda works.
One thing which I think is missing is testing the 2 other cases for `return`. According to what I understand the lambda can return 3 different things:
- the new list of jobs in prod if different than what we had.
- 'Difference in jobs found but this is not production so no deployment for you' if different than what we had but environment is prod.
- 'No differences in jobs found, not deploying' if no difference.
I think the 3 should be tested to make sure than the lambda works as expected. What do you think Anna? I would be keen to get other opinions from the engineers in the team on this. @Couto @SemraSh ?

Reply:
Fair point about testing for case 2 in your example.

Reply:
The convention is to just use one file indeed, unless the code/scenario is really complex, which is not the case.

We are testing a single function so we should merge those two files into a single `lever.test.js` file, where we can have `describe` blocks one for each scenario.

I agree with @GBouffard here, we have three possible outcomes:
- Returned jobs are equal. Nothing to do.
- Returned jobs are different:
  - is production: is the return value expected?
  - is not production: is the return value expected?

So it seems to me the last case is missing (I might be wrong)

Reply:
So I rejigged this a little bit - we still have two separate files, but now one is Prod and the other is Dev. Basically the function needs a isProd variable, so that needs to be passed as true / false, so it's easier to have two separate files with different scripts. If you have any suggestions how to make it work in a single file, I'm all ears